lib/Conversion/HandshakeToFIRRTL/HandshakeToFIRRTL.cpp

```diff
@@ -1900,6 +1954,20 @@ void HandshakeBuilder::buildDataBufferLogic(Value predValid, Value validReg,
   }
 }
 
+/// Connects src to all elements of dest recursively. If dest isn't a bundle
+/// type, a normal connection is created.
+static void connectSrcToAllElements(ImplicitLocOpBuilder &builder, Value dest,
+                                    Value src) {
+  if (auto bundleType = dest.getType().dyn_cast<BundleType>()) {
+    for (int i = 0, e = bundleType.getNumElements(); i < e; ++i) {
+      auto field = builder.create<SubfieldOp>(dest, i);
+      connectSrcToAllElements(builder, field, src);
+    }
+  } else {
+    builder.create<ConnectOp>(dest, src);
+  }
+}
+
```

Comment:
Can we use `firrtl::emitConnect`?https://github.com/llvm/circt/blob/90ae0886aa06020d9f370118bb69dc1d3afbddcb/lib/Dialect/FIRRTL/FIRRTLUtils.cpp#L28